src/owp_snow_obs/owp_snow_obs.py

```diff
@@ -0,0 +1,255 @@
+#!/usr/bin/env python3
+
+# (C) Copyright 2019 UCAR
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+
+# Purpose: Convert OWP snow observations from CSV format to IODA netcdf format.
+# Author:
+# 2020-11-17, 2021-10-18: James McCreight
+
+# Conceptual Figure:
+# TODO: this is probably out of date... a
+# https://jointcenterforsatellitedataassimilation-jedi-docs.readthedocs-hosted.com/en/latest/_images/IODA_InMemorySchematic.png
+
+# Example testing usage:
+# ipython --pdb -c"%run
+#     owp_snow_obs.py
+#     -i $test_input/owp_snow_obs.csv -o $here/owp_snow_obs_csv.nc --thin_swe 1"
+
+from datetime import datetime
+import numpy as np
+import os
+import pandas as pd
+import pathlib
+from pathlib import Path
+import sys
+import warnings
+
+
+IODA_CONV_PATH = Path(__file__).parent/"@SCRIPT_LIB_PATH@"
+if not IODA_CONV_PATH.is_dir():
+    IODA_CONV_PATH = Path(__file__).parent/'..'/'lib-python'
+sys.path.append(str(IODA_CONV_PATH.resolve()))
+
+import ioda_conv_engines as iconv
+from collections import defaultdict, OrderedDict
+from orddicts import DefaultOrderedDict
+
+arg_parse_description = (
+    """Reads snow OWP observations in CSV files and converts
+    to IODA output files. """)
+
+# obs file name -> ioda file name
+output_var_dict = {'snow_depth_m': 'snow_depth', 'snow_water_equivalent_mm': 'swe'}
+# ioda file_name -> ioda file units
+output_var_unit_dict = {'snow_depth': 'm', 'swe': 'mm'}
+one = 1.00000000000000
+output_conversion_factor = {'snow_depth': one, 'swe': one}
+
+location_key_list = [
+    ("latitude", "float"),
+    ("longitude", "float"),
+    ("datetime", "string"), ]
+
+dim_dict = {}
+
+var_dims = {
+    'snow_depth': ['nlocs'],
+    'swe': ['nlocs'], }
+
+attr_data = {
+    'converter': os.path.basename(__file__),
+    'converter_version': 0.3,
+    'nvars': np.int32(len(var_dims)), }
+
+fill_value = 9.96921e+36
+
+
+def dummy_err(obs_values, var_name):
+    obs_errs = 0.5 * obs_values
+    return obs_errs
+
+
+def mask_nans(arr):
+    arr2 = arr.astype('float32')
+    arr2 = np.nan_to_num(arr2, nan=fill_value)
+    arr2 = np.ma.masked_where(arr2 == fill_value, arr2)
```

Comment:
Numpy masked arrays can be terribly inefficient for performance, but I do realize that they can also be quite useful. If you think that this obs type won't get into the millions of locations (for example) it might be okay to use the masked arrays. Just want to warn you about this.

Reply:
This wont be an issue. The per file ocnversion takes about 1.2s Thanks for the fair warning, though.